### PR TITLE
Deploy NSG and VM public IPs when internal standard LB selected

### LINKED ISSUE #308: Custom script execution failing
Hi,
following template deployment throws an exception while executing a custom script extension while provisioning a VM, are you able to tell me what is trying to connect to please?

Enable failed: processing file downloads failed: failed to download file[0]: failed to download file: http request failed: Get [REDACTED] dial tcp 151.101.208.133:443

Command executed:
$template_base_uri="https://raw.githubusercontent.com/elastic/azure-marketplace"
$template_version="7.3.1"

az group deployment create --resource-group "dev-elk"  --template-uri $template_base_uri/$template_version/src/mainTemplate.json  --parameters _artifactsLocation=$template_base_uri/$template_version/src/ esVersion=7.3.1 esClusterName=elasticsearch vmDataDiskCount=1 dataNodesAreMasterEligible=Yes adminUsername=kaiser adminPassword=********** securityBootstrapPassword=*********securityAdminPassword=******** securityKibanaPassword=******** securityLogstashPassword=********** securityBeatsPassword=****** securityApmPassword=********* securityRemoteMonitoringPassword=*********** vmDataNodeCount=2 xpackPlugins=No logstash=Yes vmLogstashCount=2 loadBalancerType=internal loadBalancerInternalSku=Standard --vNetName elk-vnet --debug

### PR DESCRIPTION
**Pull request by russcam**
This PR deploys a Network Security Group when a `Standard` SKU internal Load Balancer is selected and the loadBalancerType is `internal`. Each VM in ther backend pool is also dynamically assigned a Standard SKU public IP address to allow outbound traffic, which is prevented by default when a Standard SKU load balancer is in front of the VMs.

Fixes #308

### CODE REVIEW

**Comment by russcam** [comment]
before merging, run

```sh
npm run links -- --branch master
```

File: src/partials/vm.json [hunk 1]
```diff
@@ -63,14 +63,40 @@
       }
     },
     "osProfile": "[variables(concat(parameters('vm').shared.credentials.authenticationType, '_osProfile'))]",
-    "externalNsgs": [
-      { },
-      { "networkSecurityGroup" : { "id": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('vm').externalNsg)]" } }
+    "publicIpName": "[concat(variables('namespace'), parameters('index'), '-ip')]",
+    "nsgIpConfigs": [
+      {},
```
Comment (Mpdreamz) [inline src/partials/vm.json]: This was there before but why the empty object?

Comment (Mpdreamz) [reply src/partials/vm.json]: Ahh, the index dispatch below explains it :)

**Comment by russcam** [comment]
> LGTM, tad sad this is something we even need to consider and does not work out of the box though.

Agreed

**Pull request merged**